src/adapter/src/coord.rs

```diff
@@ -351,6 +352,15 @@ pub struct PeekStageFinish {
     pub typ: RelationType,
 }
 
+/// An enum describing which cluster to run a statement on.
```

Comment:
nit: can you provide context here for where we'd use this? e.g. something like "if a query depends only on system tables, we might automatically run it on the introspection cluster to benefit from indexes that exist there.